sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregates.scala

```diff
@@ -67,4 +67,8 @@ object RemoveRedundantAggregates extends Rule[LogicalPlan] with AliasHelper {
 
     upperHasNoDuplicateSensitiveAgg && upperRefsOnlyDeterministicNonAgg
   }
+
+  private def isDuplicateSensitive(ae: AggregateExpression): Boolean = {
+    !ae.isDistinct && !EliminateDistinct.isDuplicateAgnostic(ae.aggregateFunction)
```

Comment:
Why don't you add `ae.isDistinct` to `isDuplicateAgnostic`?

Reply:
`EliminateDistinct.isDuplicateAgnostic` looks at `AggregateFunction`,  `isDistinct` is a property of a `AggregateExpression`.

Reply:
you could just pass `AggregateFunction` ..

Reply:
That would break `EliminateDistinct`.

`EliminateDistinct.isDuplicateAgnostic` checks if it could remove distinct from the `AggregateExpression`.
Currently it is "if the `af` is `max`, `min`.. , then remove `isDistinct`. You proposition would change it to "if `ae` has `isDistinct`, then remove `isDistinct`.

Reply:
But that's what we're already doing:

```scala
object EliminateDistinct extends Rule[LogicalPlan] {
    ...
    case ae: AggregateExpression if ae.isDistinct && isDuplicateAgnostic(ae.aggregateFunction) =>
      ae.copy(isDistinct = false)
    ...
```

Reply:
If aggregate expr is distinct, it would make sense to consider it DuplicateAgnostic .. ?

Reply:
In`EliminateDistinct` the rule is `ae.isDistinct && isDuplicateAgnostic(ae.aggregateFunction)`
In `RemoveRedundantAggregates` the rule is `!ae.isDistinct && !isDuplicateAgnostic(ae.aggregateFunction) = !(ae.isDistinct || isDuplicateAgnostic(ae.aggregateFunction))`.

These are not the same rule (`&&` vs `||`).

Reply:
Ah, okay I got what you mean

Reply:
okay make sense 👍